lib/oauth.js

```diff
@@ -147,27 +148,34 @@ exports.v2 = function (settings) {
         if (!request.query.code) {
             var nonce = Cryptiles.randomString(22);
             query = Hoek.clone(settings.providerParams) || {};
-            protocol = request.connection.info.protocol;
-            if (settings.forceHttps) {
-                protocol = 'https';
-            }
-            query.client_id = settings.clientId;
-            query.response_type = 'code';
-            query.redirect_uri = internals.location(request, protocol, settings.location);
-            query.state = nonce;
 
-            var scope = settings.scope || settings.provider.scope;
-            if (scope) {
-                query.scope = scope.join(settings.provider.scopeSeparator || ' ');
-            }
+            // Validate runtime query params
+            return Joi.validate(request.query, settings.runtimeQuerySchema, function (err, value) {
```

Comment:
I know it's the default but can you explicitly state { allowUnknown: false } in the Joi.validate options.